### ISSUE TITLE
std/node: add net module

### ISSUE THREAD

**Issue by EvanHahn** [denoland/deno#5385]
For completeness, we should add a polyfill for [Node's net module](https://nodejs.org/api/net.html).

**Comment by ebebbington** [comment]
Is it acceptable for a std/node polyfill to only implement part of the module? eg just `net.Socket` is implemented and there's a notice in the readme saying "Only `.Socket` is implemented at this moment in time"

I might try work on this if no one else is?

**Comment by guybedford** [comment]
@ebebbington yes it's absolutely fine to just create a partial implementation and this is what is done for the browser and in Deno already for many cases, often with just a stub function with a todo marked. A contribution would be most welcome.

**Comment by ebebbington** [comment]
Alright i'll try work on it when i can, but i'll note that if anyone else is wishing to tackle this issue, feel free as i don't have all the spare time in the world just lying around :P (ie it will take some time)

**Comment by guybedford** [comment]
This is open source, that is assumed :)
